backend/router.js

```diff
@@ -11,9 +12,68 @@ app.use(cors())
 app.use(express.json())
 
 const PORT = process.env.PORT || 3000
-console.log('Port:', PORT)
 
 
+const groqInstance = new groq({ apiKey: process.env.GROQ_API_KEY });
+
+app.post('/api/chat', async (req, res) => {
+    const { message, context,lastQuestionCheck, prevIsFollowUp} = req.body;
+
+    let lastQuestion = lastQuestionCheck=="quit"? true: false
+    try {
+        const doFollowUp = Math.random()>.65 ? true: false;
+        if(doFollowUp){lastQuestion= false;}
+        
+        if(prevIsFollowUp){lastQuestion=false;}
+        const response = await groqInstance.chat.completions.create({
+            messages: [
+                {
+                    role: "user",
+                    
+                    content: doFollowUp && !lastQuestion ? `You are a interviewer conducting a behavioral interview. Make sure to talk in the first person as if this was a normal conversation 
```

Comment:
maybe we can separate these into a separate strings file? would make it easier to edit the prompt in the future separately from the rest of the code (if you want to tune it further for ex)